[HttpClient] Add `HttpOptions::addHeader` method (#19291)
symfony/symfony-docs: http_client.rst

    $this->client = $client->withOptions(
        (new HttpOptions())
            ->setBaseUri('https://...')

>             // Set headers replace *all* headers at once, and deletes the headers you do not provide

Reviewer: ```suggestion
            // setHeaders() replaces *all* headers at once, and deletes the headers you do not provide
```

            ->setHeaders(['header-name' => 'header-value'])

>             // Alternatively, add or replace only one header to an existing HttpOptions object.

Reviewer: ```suggestion
            // add or replace a single header using addHeader()
```

            ->toArray()
    );

.. tip::

>     Be aware that calling :method:`Symfony\\Component\\HttpClient\\HttpOptions::setHeaders` overrides all previous headers.

Reviewer: I would update the code example above to call both the `setHeaders()` and `setHeader()` method and add inline comments explaining the difference.

Author: I've made a change in that direction. Tell me if there is anything I can improve further ;)
